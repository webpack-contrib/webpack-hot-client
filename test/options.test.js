const { createServer } = require('http');

const getOptions = require('../lib/options');

describe('options', () => {
  test('defaults', () => {
    const options = getOptions({});
    expect(options).toMatchSnapshot({
      stats: {
        context: expect.stringMatching(/(webpack-hot-client|project)$/),
      },
    });
  });

  test('altered options', () => {
    const altered = {
      allEntries: true,
      autoConfigure: false,
      host: {
        client: 'localhost',
        server: 'localhost',
      },
      hmr: false,
      https: true,
      logLevel: 'trace',
      logTime: true,
      port: 0,
      reload: false,
      send: {
        errors: false,
        warnings: false,
      },
      // this property is tested later
      // server: null,
      stats: {
        context: '/',
      },
      validTargets: ['batman'],
      // we pass this to force the log instance to be unique, to assert log
      // option differences
      test: true,
    };
    const options = getOptions(altered);
    // console.log(JSON.stringify(options, null, 2));
    expect(options).toMatchSnapshot();
  });

  test('throws on invalid options', () => {
    const t = () => getOptions({ host: true });
    expect(t).toThrow();
  });

  test('throws if host.client is missing', () => {
    const t = () => getOptions({ host: { server: 'localhost' } });
    expect(t).toThrow();
  });

  test('throws if host.server is missing', () => {
    const t = () => getOptions({ host: { client: 'localhost' } });
    expect(t).toThrow();
  });

  test('reject non-http.Server server', () => {
    const server = {};
    const t = () => getOptions({ server });
    expect(t).toThrow();
  });

  test('reject non-running server', () => {
    const server = createServer();
    const t = () => getOptions({ server });
    expect(t).toThrow();
  });
});
