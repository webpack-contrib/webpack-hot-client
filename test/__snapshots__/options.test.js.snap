// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options altered options 1`] = `
Object {
  "allEntries": true,
  "autoConfigure": false,
  "hmr": false,
  "host": Object {
    "client": "localhost",
    "server": "localhost",
  },
  "https": true,
  "log": LogLevel {
    "currentLevel": 0,
    "debug": [Function],
    "error": [Function],
    "info": [Function],
    "log": [Function],
    "methodFactory": PrefixFactory {
      "options": Object {
        "level": [Function],
        "name": [Function],
        "template": "[{{time}}] {{level}} [90mï½¢{{name}}ï½£[39m: ",
        "time": [Function],
      },
      Symbol(a log instance): [Circular],
      Symbol(valid log levels): Object {
        "DEBUG": 1,
        "ERROR": 4,
        "INFO": 2,
        "SILENT": 5,
        "TRACE": 0,
        "WARN": 3,
      },
    },
    "name": "hot",
    "options": Object {
      "factory": null,
      "level": "trace",
      "name": "hot",
      "prefix": Object {
        "level": [Function],
        "template": "[{{time}}] {{level}} [90mï½¢{{name}}ï½£[39m: ",
      },
      "timestamp": true,
      "unique": true,
    },
    "trace": [Function],
    "type": "LogLevel",
    "warn": [Function],
  },
  "logLevel": "trace",
  "logTime": true,
  "port": 0,
  "reload": false,
  "send": Object {
    "errors": false,
    "warnings": false,
  },
  "server": null,
  "stats": Object {
    "context": "/",
  },
  "test": true,
  "validTargets": Array [
    "web",
    "batman",
  ],
  "webSocket": Object {
    "host": "localhost",
    "port": 0,
  },
}
`;

exports[`options defaults 1`] = `
Object {
  "allEntries": false,
  "autoConfigure": true,
  "hmr": true,
  "host": Object {
    "client": "localhost",
    "server": "localhost",
  },
  "https": false,
  "log": LogLevel {
    "currentLevel": 2,
    "debug": [Function],
    "error": [Function],
    "info": [Function],
    "log": [Function],
    "methodFactory": PrefixFactory {
      "options": Object {
        "level": [Function],
        "name": [Function],
        "template": "{{level}} [90mï½¢{{name}}ï½£[39m: ",
        "time": [Function],
      },
      Symbol(a log instance): [Circular],
      Symbol(valid log levels): Object {
        "DEBUG": 1,
        "ERROR": 4,
        "INFO": 2,
        "SILENT": 5,
        "TRACE": 0,
        "WARN": 3,
      },
    },
    "name": "hot",
    "options": Object {
      "factory": null,
      "level": "info",
      "name": "hot",
      "prefix": Object {
        "level": [Function],
        "template": "{{level}} [90mï½¢{{name}}ï½£[39m: ",
      },
      "timestamp": false,
      "unique": true,
    },
    "trace": [Function],
    "type": "LogLevel",
    "warn": [Function],
  },
  "logLevel": "info",
  "logTime": false,
  "port": 0,
  "reload": true,
  "send": Object {
    "errors": true,
    "warnings": true,
  },
  "server": null,
  "stats": Object {
    "context": StringMatching /\\(webpack-hot-client\\|project\\)\\$/,
  },
  "test": false,
  "validTargets": Array [
    "web",
  ],
  "webSocket": Object {
    "host": "localhost",
    "port": 0,
  },
}
`;
